streams/src/test/java/org/apache/kafka/streams/processor/internals/TaskManagerTest.java

```diff
@@ -756,7 +725,7 @@ public void setCommitNeeded() {
 
         @Override
         public boolean commitNeeded() {
-            return commitNeeded;
+            return active && commitNeeded;
```

Comment:
This is the fix for `StateMachineTask` -- it should behave the same way as an actual active or standby task